src/Input/Expression.re

```diff
@@ -0,0 +1,33 @@
+type t =
+  | Variable(string)
+  | And(t, t)
+  | Or(t, t)
+  | Not(t);
+
+let rec show = (v: t) =>
+  switch (v) {
+  | Variable(s) => Printf.sprintf("Variable(%s)", s)
+  | And(v1, v2) => Printf.sprintf("And(%s, %s)", show(v1), show(v2))
+  | Or(v1, v2) => Printf.sprintf("Or(%s, %s)", show(v1), show(v2))
+  | Not(v) => "!" ++ show(v)
+  };
+
+let evaluate = (v: t, getValue) => {
+  let rec f = v =>
```

Comment:
`loop` `aux` and `helper` are common names for internal recursive function that basically just do the work. `evaluate'` might also be a possibility, but `f` is a bit too non-descriptive I think 😆

Reply:
Thanks for the feedback! Makes sense - I went with `eval` which hopefully is a bit more readable.